public/js/app/leaderboards.js

```diff
@@ -51,185 +39,130 @@ function leaderboardUpdate(data) {
   });
 
   //determines the current page, whether to add or substract the missing players in case we pressed next or previous then it will add or substract players
-  let playerIndex = (data.length - 1) - (pageNumber * 100) - (addNextPlayer * plusOrMinusPlayerList); //- addNextPlayer;
-  let next100Players = data.length - 100 - (pageNumber * 100) - (addNextPlayer * plusOrMinusPlayerList);
+  let playerIndex = (playerList.length - 1) - (pageNumber * 100); //- addNextPlayer;
+  let next100Players = playerList.length - 101 - (pageNumber * 100);
   for (playerIndex; playerIndex > next100Players; playerIndex--) {
-    
-    
-    
-    
-    
-    //console.log(`playerIndex is ${playerIndex}`);
-    //console.log(`data.length minus missing players is ${data.length - missingPlayers}`);
-    //console.log(` for loop data.length is ${data.length - 100 - (pageNumber * 100)}`)
-    
 
     if (playerIndex <= 0) {
-      //console.log('I broke');
-      playerIndex -= (playerIndex) * -1;
-    } else {
-      console.log(` We didn't break! `);
+      console.log('I broke');
     }
     //Switches background color for even and uneven
-    if (currentDate >= data[playerIndex].date) {
-      next100Players--;
-      addNextPlayer++;
-      previousMissingPlayer++;
-      //console.log(`${data[playerIndex].label} didn't play recently`); 
-      
-      
+    if (backgroundColor === 'greyLeaderboard') {
+      backgroundColor = 'darkGreyLeaderboard';
```

Comment:
1) Why ?
2) Use constants, no magics

Reply:
but I need the background color to change for even and uneven players. If I make it a constant, wouldnt it stay constantly in one color?

Reply:
Use even and odd CSS rules ! https://www.w3.org/Style/Examples/007/evenodd.en.html 🙂
No need for javascript for that

Reply:
Oh this was very useful as well, thanks alot!